docs/proposals/standardize-ring-monitoring.md

```diff
@@ -0,0 +1,166 @@
+
+- MimirAlertmanagerRingCheckFailing alerting rule is based on metric `cortex_alertmanager_ring_check_errors_total` only available for alertmanager
+
+  - Only apply to alertmanager ring
+  - Expose ring usability
+  - Does not alert about an unused broken ring.
+  - This method needs to be reimplemented for every ring
+
+- MimirRulerFailedRingCheck alerting rule is based on metric `cortex_ruler_ring_check_errors_total` only available for ruler
+  - Only apply to ruler ring
+  - Give an interesting point a view about the usability of the ring.
+  - Does not alert about an unused broken ring.
+  - This method needs to be reimplemented for every ring
+
+## Goals
+
+- Monitor all rings used inside Mimir software
+- Give both a view of the state of the ring and of its usability
+- Standardize monitoring of rings
+- Automatically adapt to presents rings are some of thems are optionals
+
+## Non-Goals (optional)
+
+- Improve monitoring of other dskit based tools(loki/tempo/pyroscope) even if it could be a nice addition.
+
+## Proposals
+
+### Proposal 0: Do nothing
+
+Current solution has proven to be working. The most critical ring (a.k.a ingester) is monitored.
+
+**Pros:**
+
+- Nothing to be done
+
+**Cons:**
+
+- Not all mimir rings are monitored
+- Not homogenous among all rings
+- Alert does not taking into account availability zone
+- No indication on the ring state, only if there are unhealthy nodes and only for the ingester ring
+
+### Proposal 1: Make alert MimirIngesterUnhealthy more generic
+
+Rename the probe `MimirIngesterUnhealthy` to `MimirRingUnhealthy` and change it's expression to
+
+    min by (cluster, namespace, name) (cortex_ring_members{state="Unhealthy"}) > 0
+
+**Pros:**
+
+- Really easy to implement as all needed ressources are already available inside mimir.
+- All rings are monitored.
+- Monitoring is partially consistent among all rings inside the cluster (except for the two additional probes on rings ruler and alertmanager)
+- Monitoring adapts automatically to the rings activated inside mimir.
+
+**Cons:**
+
+- Probe are not taking into account zone-awareness
+- Probe do not fully refelect the state of the ring (usable or not)
+- Probe do not reflect the usage of the ring by the client except for the two additional probes on rings ruler and alertmanager
+
+### Proposal 2: Copy ruler and alertmanager monitoring for other rings
+
+Proposal 1 +
+
+Implements metric cortex\_<ring_name>\_ring_check_errors_total for all rings in mimir.
+
+Add as many alerting rules as there are rings inside mimir by using the following expression:
+
+    sum by (cluster, namespace, job) (rate(cortex_<ring_name>_ring_check_errors_total[1m]))
+
+**Pros:**
+
+- All rings are monitored.
+- Monitoring is consistent among all rings inside the cluster.
+
+**Cons:**
+
+- Probe only reflects usage of the ring by the client but not the ring' state.
```

Comment:
interesting point. Do we want to alert on the state or do we want to when something is wrong? I think we want to [alert on symptoms, not on causes](https://cloud.google.com/blog/topics/developers-practitioners/why-focus-symptoms-not-causes), so I'd say that we want to alert when ring operations are failing. I like this proposal